src/components/page-create-application/index.tsx

```diff
@@ -59,24 +59,19 @@ function PageCreateApplication({
   useEffect(() => {
     if (!visibleScrim) return;
 
-    switch (creationState) {
-      case RequestState.FAILURE:
-        scrollToPosition(
-          containerRef.current,
-          0,
-          containerRef.current?.scrollHeight
-        );
-        break;
-      case RequestState.SUCCESS:
-        if (
-          creationResponse.applicationRegistration &&
-          !creationResponse.warnings
-        ) {
-          scrollToPosition(containerRef.current, 0, 0);
-        }
-        break;
-      default:
-        break;
+    if (creationState === RequestState.FAILURE) {
+      scrollToPosition(
+        containerRef.current,
+        0,
+        containerRef.current?.scrollHeight
+      );
+    } else if (creationState === RequestState.SUCCESS) {
+      if (
+        creationResponse.applicationRegistration &&
+        !creationResponse.warnings
+      ) {
+        scrollToPosition(containerRef.current, 0, 0);
+      }
```

Comment:
Reduce complexity
```suggestion
    if (creationState === RequestState.FAILURE) {
      scrollToPosition(
        containerRef.current,
        0,
        containerRef.current?.scrollHeight
      );
      return;
    }
    if (
      creationState === RequestState.SUCCESS &&
      creationResponse.applicationRegistration &&
      !creationResponse.warnings
    ) {
      scrollToPosition(containerRef.current, 0, 0);
    }
```

Reply:
I know which pattern you are referring to, but the pattern is designed to filter unnecessary complexity in regular `if/else` chains. It does not go out of its way to remove every `else if` as well.

`The “return early” pattern is an excellent way to prevent functions from becoming confusing. However, this doesn’t mean that it can be applied every time.`
https://medium.com/swlh/return-early-pattern-3d18a41bba8

I am not an opponent of this pattern, quite the opposite, but I disagree in this case in respect of readability and code style.